algorithms/src/traits/merkle_tree.rs

```diff
@@ -32,6 +32,7 @@ pub trait MerkleParameters: Send + Sync + Clone {
 
     /// Returns the hash of a given leaf.
     fn hash_leaf<L: ToBytes>(&self, leaf: &L, buffer: &mut [u8]) -> Result<<Self::H as CRH>::Output, MerkleError> {
+        debug_assert_eq!(buffer.len(), Self::H::INPUT_SIZE_BITS / 8);
```

Comment:
To prevent such overallocation of the Merkle tree buffer, a `debug_assert_eq` is left here.